src/ChunkMap.h

```diff
@@ -492,27 +431,26 @@ class cChunkMap
 		}
 	};
 
-	typedef std::list<cChunkLayer *> cChunkLayerList;
-
-	typedef std::list<cChunkStay *> cChunkStays;
-
-	/** Finds the cChunkLayer object responsible for the specified chunk; returns nullptr if not found. Assumes m_CSLayers is locked. */
-	cChunkLayer * FindLayerForChunk(int a_ChunkX, int a_ChunkZ);
+	struct ChunkCoordinate
+	{
+		bool operator() (const ChunkCoordinate & a_Lhs, const ChunkCoordinate & a_Rhs) const
```

Comment:
Wait, what? Why does a coordinate pair define call-operator, and why does the call operator compare two coordinates? This is very confusing at first look. Isn't it possible to make the comparison function a static one, and use it directly in the map's definition as the third template param?